src/eventloop.jl

```diff
@@ -52,7 +57,63 @@ function waitloop()
     end
 end
 
-const idle_delay = Ref(0.5)
-function set_idle_delay(delay_secs::Float64=0.5)
-    idle_delay[] = delay_secs
+#=
+    Hold Channel Interface
+    This allows user processes to register and deregister
+    channels that allow user code to hold IJulia from sending
+    the idle status message until signaled.  This would have
+    been prettier if there were level-triggered events in
+    Julia base, but this will do okay for now.
+
+    Users should reset their own channels as IJulia will not
+    reach into their channel state manually unless there is a
+    timeout, in which case a temporary variable is put onto the channel
+    buffer and removed after the wait process.
+    See wait_multi_with_timeout()
+=#
+# Interact.jl band-aid compatibility code.
+const interact_min_delay = 0.25
+
+# Channel constants
+const hold_channels = Array{Channel{Bool},1}()
+const hold_channels_timeout_default = 3
+
+function add_hold_channel(channel::Channel{Bool})
+    if !(channel in hold_channels)
+        hold_channels = [hold_channels..., channel]
+    end
+end
+
+function remove_hold_channel(channel::Channel{Bool})
+    if channel in hold_channels
+        filter!(e->e∉[channel],hold_channels)
+    end
 end
+
+function wait_multi_with_timeout(channels::Array{Channel{Bool},1}, timeout::Number)
+    notification = Channel{Bool}(1)
+    #=
+        This async routine will manually fulfill any channels (aka semaphores)
+        that don't come through before the timeout.  Doing this because someone
+        will inevitably not fulfill the semaphores that they register, and
+        thus this will take care of us, at least for the short term.
```

Comment:
I'm not clear on what the timeout is for.   User code is free to do infinite loops, or sleep forever, so why would we try to catch bugs where they failr to release the semaphore?